### PR TITLE
[AP-762][Feat] Support transformation of fields in json-type columns

### PR DESCRIPTION
**Pull request by Samira-El**
## Context

Currently, we can apply transformations on various columns of simple types such as number, strings, dates ...etc. However, we don't have the same feature to transform certain fields nested in a json-type column.

This PR is add this capability, it makes use of XPaths as a way to point to one or multiple fields in a JSON object and transforming it. 

### Example

Assume you have a stream named `my_cool_db-users` which has an field of type object which contains metadata about users. 
The tap produces singer records such as these:
```
{
            "id": 100,
            "first_name": "John",
            "last_name": "Smith",
            "metadata": {"user": {"address": "Vereeni 23" , "zip_code": "362294"}, "browser": "Firefox", "platform": "android"}
        }
```

Given that this streams contains sensitive data that need to be transformed, you now can transform the specified fields inside the `metadata` object like so:

```
"transformations": [
   {
        "field_id": "metadata",
        "tap_stream_name": "my_cool_db-users",
        "type": "MASK-HIDDEN",
        "field_paths": ["user/address", "user/zip_code"]
   }
]
```

### CODE REVIEW

File: tests/unit/test_transform.py [hunk 1]
```diff
@@ -137,3 +137,77 @@ def test_conditions(self):
...
+            # Column to transform:
+            "col_4",
+            # Transform method:
+            "MASK-HIDDEN",
+            # Conditions when to transform:
+            None,
+            ['info/last_name', 'info/first_name']
+        )
+
+        self.assertEqual(expected_value, return_value)
```
Comment (amofakhar) [inline tests/unit/test_transform.py:213]:
```suggestion
        self.assertDictEqual(expected_value, return_value)
```

File: tests/unit/test_transform.py [hunk 2]
```diff
@@ -137,3 +137,77 @@ def test_conditions(self):
...
+            # Column to transform:
+            "col_4",
+            # Transform method:
+            "MASK-HIDDEN",
+            # Conditions when to transform:
+            None,
+            ['info/last_name']
+        )
+
+        self.assertEqual(expected_value, return_value)
```
Comment (amofakhar) [inline tests/unit/test_transform.py:163]:
```suggestion
        self.assertDictEqual(expected_value, return_value)
```

File: tests/unit/test_transform.py [hunk 3]
```diff
@@ -137,3 +137,77 @@ def test_conditions(self):
...
+            # Transform method:
+            "MASK-HIDDEN",
+            # Conditions when to transform:
+            [
+                {'column': 'col_2', 'equals': "passwordHash"},
+            ],
+            ['info/last_name']
+        )
+
+        self.assertEqual(expected_value, return_value)
```
Comment (amofakhar) [inline tests/unit/test_transform.py:189]:
```suggestion
        self.assertDictEqual(expected_value, return_value)
```

**Review by amofakhar — COMMENTED** [review]
Looks good to me!

**Comment by Samira-El** [comment]
> Looks good to me!

Thanks @amofakhar , can you approve?

**Review by amofakhar — APPROVED** [review]

**Pull request merged**